### PR TITLE
Bug 1482475 - Add extensive testing framework

### PR DESCRIPTION
**Pull request by dylanwh**
This doesn't test everything I want to -- but it adds the right framework for writing tests more easily.

### CODE REVIEW

File: extensions/PhabBugz/lib/Feed.pm [hunk 1]
```diff
@@ -382,7 +382,7 @@ sub process_revision_change {
 
     # Pre setup before making changes
     my $old_user = set_phab_user();
-    my $bug = Bugzilla::Bug->new({ id => $revision->bug_id, cache => 1 });
+    my $bug = $revision->bug;
```
Comment (dylanwh) [inline extensions/PhabBugz/lib/Feed.pm]: @dklawren Is this a good / safe thing to fix here?

File: extensions/PhabBugz/lib/Feed.pm [hunk 2]
```diff
@@ -688,7 +688,7 @@ sub process_new_user {
         # that are connected to revisions
         f11 => 'attachments.filename',
         o11 => 'regexp',
-        v11 => '^phabricator-D[[:digit:]]+-url[[.period.]]txt$',
+        v11 => '^phabricator-D[[:digit:]]+-url.txt$',
```
Comment (dylanwh) [inline extensions/PhabBugz/lib/Feed.pm]: [[.period.]] is a mysql-ism that won't work with sqlite. Technically the regex is more open now, but I think it won't hurt.

File: extensions/PhabBugz/lib/Feed.pm (hunk 1 above)
Comment (dklawren) [reply extensions/PhabBugz/lib/Feed.pm]: Yeah pretty much the same thing.

File: extensions/PhabBugz/lib/Feed.pm (hunk 2 above)
Comment (dklawren) [reply extensions/PhabBugz/lib/Feed.pm]: works for me

**Pull request merged**